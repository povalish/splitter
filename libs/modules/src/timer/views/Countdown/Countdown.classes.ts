import { cva } from 'class-variance-authority';

//
//

export const countdown = cva('text-[114px] text-center font-bold text-colorText');
