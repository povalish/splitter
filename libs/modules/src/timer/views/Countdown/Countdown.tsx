import { formatSeconds } from '@modules/timer/utils/formatSeconds';
import React from 'react';

//
//

interface ICountdown {
  seconds: number;
}

export const Countdown: React.FC<ICountdown> = ({ seconds }) => {
  const time = formatSeconds(seconds);
  return (
    <h1 className='text-[114px] text-center font-bold'>
      {time.minutes}:{time.seconds}
    </h1>
  );
};
