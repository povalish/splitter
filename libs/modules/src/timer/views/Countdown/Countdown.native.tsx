import React from 'react';
import { Text } from 'react-native';
import { formatSeconds } from '@modules/timer/utils/formatSeconds';
import { countdown } from './Countdown.classes';
import { IStyled } from '@ui-kit/types/Styled';

//
//

interface ICountdown extends IStyled {
  seconds: number;
}

export const Countdown: React.FC<ICountdown> = ({ seconds, className }) => {
  const time = formatSeconds(seconds);
  return (
    <Text className={countdown({ className })}>
      {time.minutes}:{time.seconds}
    </Text>
  );
};
