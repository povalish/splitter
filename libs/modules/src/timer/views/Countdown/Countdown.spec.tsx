import React from 'react';
import { render, screen } from '@testing-library/react';
import { Countdown } from './Countdown';

//
//

describe('TIMER MODULES - VIEWS - Countdown component', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Countdown seconds={120} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should format seconds', () => {
    render(<Countdown seconds={120} />);
    expect(screen.getByText('02:00'));
  });
});
