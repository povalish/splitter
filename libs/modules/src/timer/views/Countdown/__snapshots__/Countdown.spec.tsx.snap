// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TIMER MODULES - VIEWS - Countdown component should render successfully 1`] = `
<body>
  <div>
    <h1
      class="text-[114px] text-center font-bold"
    >
      02
      :
      00
    </h1>
  </div>
</body>
`;
